src/Numerics/ODESolvers/LowStorageRungeKutta3NMethod.jl

```diff
@@ -0,0 +1,273 @@
+            [slow_δ, slow_rv_dQ, slow_scaling])
+
+Use the 3N low storage Runge--Kutta method `lsrk3n` to step `Q` forward in time
+from the current time `time` to final time `time + getdt(lsrk3n)`.
+
+If the optional parameter `slow_δ !== nothing` then `slow_rv_dQ * slow_δ` is
+added as an additionall ODE right-hand side source. If the optional parameter
+`slow_scaling !== nothing` then after the final stage update the scaling
+`slow_rv_dQ *= slow_scaling` is performed.
+"""
+function dostep!(
+    Q,
+    lsrk3n::LowStorageRungeKutta3N,
+    p,
+    time,
+    slow_δ = nothing,
+    slow_rv_dQ = nothing,
+    in_slow_scaling = nothing,
+)
+    dt = lsrk3n.dt
+
+    RKA, RKB, RKC = lsrk3n.RKA, lsrk3n.RKB, lsrk3n.RKC
+    rhs!, dQ, dR = lsrk3n.rhs!, lsrk3n.dQ, lsrk3n.dR
+
+    rv_Q = realview(Q)
+    rv_dQ = realview(dQ)
+    rv_dR = realview(dR)
+    groupsize = 256
+
+    rv_dR .= -0
+    for s in 1:length(RKC)
+        rhs!(dQ, Q, p, time + RKC[s] * dt, increment = true)
+
+        slow_scaling = nothing
+        if s == length(RKC)
+            slow_scaling = in_slow_scaling
+        end
+        # update solution and scale RHS
+        event = Event(array_device(Q))
+        event = update!(array_device(Q), groupsize)(
+            rv_dQ,
+            rv_dR,
+            rv_Q,
+            RKA[s % length(RKC) + 1, 1],
+            RKA[s % length(RKC) + 1, 2],
+            RKB[s, 1],
+            RKB[s, 2],
+            dt,
+            slow_δ,
+            slow_rv_dQ,
+            slow_scaling;
+            ndrange = length(rv_Q),
+            dependencies = (event,),
+        )
+        wait(array_device(Q), event)
+    end
+end
+
+@kernel function update!(dQ, dR, Q, rka1, rka2, rkb1, rkb2, dt, slow_δ, slow_dQ, slow_scaling)
+    i = @index(Global, Linear)
+    @inbounds begin
+        if slow_δ !== nothing
+            dQ[i] += slow_δ * slow_dQ[i]
+        end
+        Q[i] += rkb1 * dt * dQ[i] + rkb2 * dt * dR[i]
+        dR[i] += rka2 * dQ[i]
+        dQ[i] *= rka1
+        if slow_scaling !== nothing
+            slow_dQ[i] *= slow_scaling
+        end
+    end
+end
+
+"""
+    LS3NRK44Classic(f, Q; dt, t0 = 0)
+
+This function returns a [`LowStorageRungeKutta3N`](@ref) time stepping object
+for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f(Q, t)
+```
+
+with the required time step size `dt` and optional initial time `t0`.  This
+time stepping object is intended to be passed to the `solve!` command.
+
+This uses the classic fourth-order, 4-stage, Runge--Kutta scheme
+in the low-storage implementation of Blum (1962)
+
+### References
+
+   @article {Blum1962,
+       title = {A Modification of the Runge-Kutta Fourth-Order Method}
+       author = {Blum, E. K.},
+       journal = {Mathematics of Computation},
+       volume = {16},
+       pages = {176-187},
+       year = {1962}
+    }
+"""
+function LS3NRK44Classic(
```

Comment:
Other places we have used the convention of using the author's names in the type.

I'm fine either way, just thought I'd point this out.

Reply:
I could change this, but thought the original scheme was from Runge + Kutta in the early 1900s, and was known as this
"classic" RK-4. The Blum 1962 paper is more relevant for the implementaion (3-N storage) rather than the property of the scheme.

Reply:
Sounds good.